microsoft/testsuites/nested/common.py

```diff
@@ -165,17 +169,20 @@ def hyperv_remove_nested_vm(
     file_path = f"{HYPER_IMAGE_FOLDER}\\{image_name}"
     hyperv = host.tools[HyperV]
 
-    # Delete VM
-    hyperv.delete_vm(name)
+    try:
+        # Delete VM
+        hyperv.delete_vm(name)
 
-    # delete image
-    host.tools[Rm].remove_file(file_path)
+        # delete image
+        host.tools[Rm].remove_file(file_path)
 
-    # delete nat network
-    hyperv.delete_nat_networking(switch_name, nat_name)
+        # delete nat network
+        hyperv.delete_nat_networking(switch_name, nat_name)
 
-    # enable port forwarding
-    hyperv.delete_port_forwarding(nat_name)
+        # enable port forwarding
+        hyperv.delete_port_forwarding(nat_name)
+    except Exception as e:
+        raise BadEnvironmentStateException(f"Failed to cleanup environment: {e}")
```

Comment:
This method is called in final already. So, if it raises another exception, it will overwrite the original exception. So if there is any error, just dump an error log and mark the environment is dirty. Not to raise bad env exception.

Reply:
Sorry I didn't get this part. Did you recently add a features to mark environment as dirty? I have used `BadEnvironmentStateException` pretty much everywhere else in my code.

Reply:
Call `node.node.mark_dirty()` or `env.mark_dirty()`. 

The mark dirty happens whatever test results, pass or fail. It doesn't affect test result, just let LISA to recycle this environment, not to use by further test runs. It's used like here, the test logic is completed (pass or fail), but the clean up phase is failed.

The BadEnvironmentException happens on the test case failed due to bad environment status, and the environment should be recycled as well. It shouldn't overwrite the original failure, because raise an exception when an exception happens will hide the original exception.

Reply:
Got it! Thanks for clarifying